src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_x509.c

```diff
@@ -42,8 +42,17 @@ AppleCryptoNative_X509DemuxAndRetainHandle(CFTypeRef handle, SecCertificateRef*
 
 static void InitCertificateCopy()
 {
+#if defined(TARGET_IOS) || defined(TARGET_TVOS)
```

Comment:
```suggestion
#if defined(TARGET_IOS) || defined(TARGET_TVOS) || defined(TARGET_MACCATALYST)
```

Reply:
This is intentional. Mac Catalyst links against the macOS version of the library and needs to take the macOS branch. It will always find the `SecCertificateCopyKey` API and use it anyway.

Reply:
For more context, the unified `SecCertificateCopyKey` API is available on iOS 12.0+, macOS 10.14+, Mac Catalyst 13.0+. The older `SecCertificateCopyPublicKey` API is available on older platforms but has different signature between iOS/tvOS/watchOS and macOS/MacCatalyst. Technically the `SecCertificateCopyPublicKey` API was never present on any Mac Catalyst platform but since the .dylib files are shared with macOS it would have the same signature.

I can make a special non-dynamic version of the code for Mac Catalyst but it makes little difference in practice.

Reply:
Ah ok, the code is fine then but could use some comment explaining this :)

We could also add min iOS version checks like we have in https://github.com/dotnet/runtime/blob/dd2dd09cf65f79ed618dae54a3ee1412cd7eed7e/src/libraries/Native/Unix/System.Security.Cryptography.Native.Apple/pal_sec.c#L8

Reply:
I focused on fixing the bug that causes incorrect behavior on older iOS versions. Let's see what @bartonjs comes up with. I'll likely be forced to do at least one more round of changes so I can improve this.

Reply:
I'll submit separate PR to improve this once it's merged. It's cleaner to use the weak linking feature of CLang on Apple platforms: https://github.com/dotnet/runtime/commit/adf30b39aa15dbde4b7740604430fc8e507d2ebc

The same thing could be done for `AppleCryptoNative_SecCopyErrorMessageString` and few other things.

Reply:
More complete patch for the weak linking changes: https://github.com/dotnet/runtime/compare/d74211c...filipnavara:sscan-avail?expand=1